main.go

```diff
@@ -0,0 +1,358 @@
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(machineconfigapi.AddToScheme(scheme))
+	utilruntime.Must(upgradev1alpha1.AddToScheme(scheme))
+	utilruntime.Must(monitoringv1.AddToScheme(scheme))
+	utilruntime.Must(routev1.Install(scheme))
+	utilruntime.Must(configv1.Install(scheme))
+	utilruntime.Must(machineapi.Install(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func printVersion() {
+	log.Info(fmt.Sprintf("Operator Version: %s", version.Version))
+	log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+	log.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+	log.Info(fmt.Sprintf("Version of operator-sdk: v%v", version.SDKVersion))
+}
+
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":"+fmt.Sprintf("%d", metricsPort), "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	// Add a custom logger to log in RFC3339 format instead of UTC
+	configLog := uzap.NewProductionEncoderConfig()
+	configLog.EncodeTime = func(ts time.Time, encoder zapcore.PrimitiveArrayEncoder) {
+		encoder.AppendString(ts.UTC().Format(time.RFC3339Nano))
+	}
+	logfmtEncoder := zaplogfmt.NewEncoder(configLog)
+	logger := zap.New(zap.UseDevMode(true), zap.WriteTo(os.Stdout), zap.Encoder(logfmtEncoder))
+	logf.SetLogger(logger)
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	printVersion()
+
+	operatorNS, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		setupLog.Error(err, "unable to determine operator namespace, please define OPERATOR_NAMESPACE")
+		os.Exit(1)
+	}
+
+	// This set the sync period to 5m
+	syncPeriod := muocfg.SyncPeriodDefault
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Namespace:              operatorNS,
+		Scheme:                 scheme,
+		MetricsBindAddress:     metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "312e6264.managed.openshift.io",
+		SyncPeriod:             &syncPeriod,
+		NewClient: func(_ cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
+			return client.New(config, options)
+		},
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	// Ensure lock for leader election
+	err = leader.Become(context.TODO(), "managed-upgrade-operator-lock")
+	if err != nil {
+		setupLog.Error(err, "failed to create leader lock")
+		os.Exit(1)
+	}
```

Comment:
One small thing here.. how would you feel about changing this so that it only becomes a leader if running on-cluster? Otherwise this block prevents the ability to run locally.

Something like this should maybe work, can you think of anything that might be an issue with it?
```suggestion
	_, err = k8sutil.GetOperatorNamespace()
	if err == nil {
		err = leader.Become(context.TODO(), "managed-upgrade-operator-lock")
		if err != nil {
			setupLog.Error(err, "failed to create leader lock")
			os.Exit(1)
		}
	} else if err != k8sutil.ErrRunLocal {
		setupLog.Error(err, "cannot get operator namespace")
		os.Exit(1)
	}
```

Reply:
Thanks Matt!! I missed to fix this.. I referred to https://github.com/operator-framework/operator-sdk/blob/v0.18.x/pkg/leader/leader.go#L48-L55 and put similar logic in another commit.. Local run skips leader election now.. Let me know if this is ok with you..